RSDK-2670 Testing QOL: add generic unordered list comparison

maximpertsov opened the pull request:
Follow-up to https://github.com/viamrobotics/rdk/pull/3996 - add a generic testing functionality that verifies is two slices of comparable elements have the same elements without considering order. This ends up removing a lot of boilerplate from resource tests.

maximpertsov commented on robot/impl/robot_reconfigure_test.go:
```diff
@@ -582,9 +582,9 @@ func TestRobotReconfigure(t *testing.T) {
 		test.That(t, robot.RemoteNames(), test.ShouldBeEmpty)
-		rdktestutils.VerifySameElements(t, arm.NamesFromRobot(robot), rdktestutils.ExtractNames(armNames...))
-		rdktestutils.VerifySameElements(t, motor.NamesFromRobot(robot), rdktestutils.ExtractNames(motorNames...))
-		rdktestutils.VerifySameElements(t, base.NamesFromRobot(robot), rdktestutils.ExtractNames(baseNames...))
```
I'm confused on how this test is passing

cheukt reviewed (approved):
LGTM, thanks for doing this!

benjirewis commented on robot/impl/robot_reconfigure_test.go:
```diff
@@ -136,70 +136,41 @@ func TestRobotReconfigure(t *testing.T) {
+		rdktestutils.VerifySameElements(t, arm.NamesFromRobot(robot), rdktestutils.ExtractNames(armNames...))
+		rdktestutils.VerifySameElements(t, base.NamesFromRobot(robot), rdktestutils.ExtractNames(baseNames...))
+		rdktestutils.VerifySameElements(t, board.NamesFromRobot(robot), rdktestutils.ExtractNames(boardNames...))
+		test.That(t, camera.NamesFromRobot(robot), test.ShouldBeEmpty)
```
I assume the `utils.NewStringSet` was unnecessary here? And, `test.ShouldBeEmpty` just verifies `len == 0`?

maximpertsov replied:
correct on both counts, the `NewStringSet` conversion was not necessary, and `test.ShouldBeEmpty` uses reflection to determine if something is a slice and then checks `.Len() == 0`

benjirewis reviewed (approved):
Hugeee thank you 🙏🏻

maximpertsov commented:
@cheukt @benjirewis FYI, I made one small change post-approval: https://github.com/viamrobotics/rdk/pull/4050/commits/6bcda5ebae284b770c534aee0217ce03ada3eff9

the helper function I added has nothing to do with resources anymore, so I moved it from `testutils/resource_utils.go` to the newly-created `testutils/slices.go`.

benjirewis commented:
Sounds good to me 👍🏻
